### ISSUE TITLE
panic: queue can not be run twice

### ISSUE THREAD

**Issue by howardjohn** [istio/istio#28891]
```
2020-11-14T00:03:03.260422Z     info    klog    failed to renew lease istio-system/istio-namespace-controller-election: timed out waiting for the condition[]
2020-11-14T00:03:03.260483Z     error   klog    Failed to release lock: resource name may not be empty[]
2020-11-14T00:03:03.260492Z     info    leader election lock lost: istio-namespace-controller-election
2020-11-14T00:03:03.260499Z     error   Leader election cycle 12 lost. Trying again
2020-11-14T00:03:03.460276Z     info    klog    attempting to acquire leader lease  istio-system/istio-namespace-controller-election...[]
2020-11-14T00:03:04.860239Z     info    klog    successfully acquired lease istio-system/istio-namespace-controller-election[]
2020-11-14T00:03:04.860423Z     info    Namespace controller started
panic: queue can not be run twice

goroutine 31279 [running]:
istio.io/istio/pkg/queue.(*queueImpl).Run(0xc005a46370, 0xc014a39800)
        istio.io/istio/pkg/queue/instance.go:76 +0x365
created by istio.io/istio/pilot/pkg/serviceregistry/kube/controller.(*NamespaceController).Run
        istio.io/istio/pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go:125 +0x14c
```

On master

cc @hzxuzhonghu 

I triggered this by setting istiod CPU limit to 100m

**Comment by hzxuzhonghu** [comment]
I think this has been fixed by #25965, you are testing on an older version?

**Comment by howardjohn** [comment]
No this was running from master

**Comment by hzxuzhonghu** [comment]
>panic: queue can not be run twice

I have reverted  #25965 in #28021, and `queue can not be run twice` this is not even exist

**Comment by howardjohn** [comment]
huh. Agreed. I must have been using an old version... thanks!

**Comment by john-deng** [comment]
@howardjohn I just tested the latest release v1.8.0 with this guide: https://istio.io/latest/docs/setup/install/multicluster/multi-primary/, it has the same issue,

```bash
2020-11-20T02:14:26.790193Z	info	klog	successfully acquired lease istio-system/istio-namespace-controller-election[]
2020-11-20T02:14:26.790317Z	info	Namespace controller started
2020-11-20T02:14:26.791071Z	info	klog	successfully acquired lease istio-system/istio-validation-controller-election[]
2020-11-20T02:14:26.791194Z	info	Starting validation controller
2020-11-20T02:14:26.791228Z	info	validationController	Reconcile(enter): add event (v1, Kind=Endpoints) istio-system/istiod
2020-11-20T02:14:26.874863Z	info	Adding cluster_id=kind-cluster02 from secret=istio-system/istio-remote-secret-kind-cluster02
2020-11-20T02:14:26.875666Z	info	initializing Kubernetes credential reader for cluster kind-cluster02
2020-11-20T02:14:26.877434Z	info	Number of remote clusters: 1
2020-11-20T02:14:26.877599Z	info	ads	Full push, service accounts changed, kubernetes.default.svc.cluster.local
2020-11-20T02:14:26.877643Z	info	ads	Full push, service accounts changed, kubernetes.default.svc.cluster.local
2020-11-20T02:14:26.877663Z	info	Handle EDS endpoint: skip collecting workload entry endpoints, service kube-scheduler/kube-system has not been populated
2020-11-20T02:14:26.877670Z	info	ads	Incremental push, service kube-scheduler.kube-system.svc.cluster.local has no endpoints
2020-11-20T02:14:26.877686Z	info	Handle EDS endpoint: skip collecting workload entry endpoints, service kube-controller-manager/kube-system has not been populated
2020-11-20T02:14:26.877695Z	info	ads	Incremental push, service kube-controller-manager.kube-system.svc.cluster.local has no endpoints
2020-11-20T02:14:26.877724Z	info	Handle EDS endpoint: skip collecting workload entry endpoints, service rancher.io-local-path/local-path-storage has not been populated
2020-11-20T02:14:26.877734Z	info	ads	Incremental push, service rancher.io-local-path.local-path-storage.svc.cluster.local has no endpoints
panic: queue can not be run twice

goroutine 449 [running]:
istio.io/istio/pkg/queue.(*queueImpl).Run(0xc00141fea0, 0xc0001151a0)
	istio.io/istio/pkg/queue/instance.go:76 +0x365
istio.io/istio/pilot/pkg/serviceregistry/kube/controller.(*Controller).Run(0xc001060a80, 0xc0001151a0)
	istio.io/istio/pilot/pkg/serviceregistry/kube/controller/controller.go:640 +0xcc
created by istio.io/istio/pilot/pkg/serviceregistry/aggregate.(*Controller).Run
	istio.io/istio/pilot/pkg/serviceregistry/aggregate/controller.go:281 +0x8d
```

**Comment by hzxuzhonghu** [comment]
Thanks https://github.com/istio/istio/pull/29068

**Comment by bergemalm** [comment]
Same issue as @john-deng here now. I guess docker containers hasn't been rebuilt with the fix? @hzxuzhonghu

**Comment by hzxuzhonghu** [comment]
1.8.1 has not been released

**Comment by Sathishkunisai** [comment]
hi, Is this issue resolved yet, I am facing similar issue on 1.8.0

**Comment by bergemalm** [comment]
I didn't get the issue when upgrading to 1.8.1
